docker/package/package_generator.py

```diff
@@ -132,7 +141,7 @@ def main():
     for package_name in args.packages:
         packages.append(all_packages[package_name])
 
-    if is_source and source_archives:
+    if is_source and (source_archives or dl_sources):
```

Comment:
I think that these should be mutually exclusive.

AFAIU right now if I provide both a source archive dir and use `--launchpad-sources` then a source archive in the dir (if any) will be preferred over one to download.

This could easily be confusing in some situations, I'd prefer an easier and more predictable behavior